File: Maths/Algebra/Factorial/factorial.rb
puts "Enter a non-negative number:"
num= gets.to_i

fact=1
if (num<0)
	puts "Factorial of a non-negative number doesn't exists"
else
	i=1
	while(i<=num)
		fact=fact*i
		i+=1
	end
	
	puts "factorial of #{num} is #{fact}"
end

File: Maths/Algebra/Factorial/factorial.py
# recursive function to calculate factorial of a number
def fact(num):

    # if the number is greater than 0
    # make a recursive call to the function
    if num > 0:
        factorial = num * fact(num-1)
    
    else:
        factorial = 1
    
    # return the factorial
    return factorial

# taking an integer input
n = int(input("Enter the number: "))

# print the factorial of the number
print(fact(n))